### ISSUE TITLE
incompatible_prohibit_aapt1: Prohibit Bazel usage of aapt1

### ISSUE THREAD

**Issue by bchang** [bazelbuild/bazel#10000]
Tracking issue for the incompatible change to prohibit aapt1.

Bazel now uses aapt2 by default (https://github.com/bazelbuild/bazel/issues/6907) meaning aapt support is deprecated.

When this change goes in effect, aapt will no longer be available.

To resolve issues when migrating your app to build with aapt2, see https://developer.android.com/studio/command-line/aapt2#aapt2_changes

**Comment by dslomov** [comment]
Since the flag was not in 1.0, "migration-1.0" is an inappropriate label. Please follow https://www.bazel.build/breaking-changes-guide.html for introducing breaking changes.

**Comment by hlopko** [comment]
It's also not in 1.1, is it?

**Comment by hlopko** [comment]
cc @meisterT

**Comment by jin** [comment]
No - the commit isn't tagged with 1.1.

**Comment by changusmc** [comment]
@jin asked me to comment here instead of filing a new issue. There's still one action that is using `aapt1` and when I build my project that uses Jetpack Navigation, it hits the code path for https://github.com/bazelbuild/bazel/blob/c018f607c9b52d301841c4d0fd2d8771c711d299/src/tools/android/java/com/google/devtools/build/android/AndroidResourceValidatorAction.java#L148

I included a repro project below.

### Description of the problem / feature request:
The source of the problem stemmed building our app which has Jetpack Navigation. Running `bazel build //android/...` gives me the same error found in https://stackoverflow.com/a/57174533/3338374

```
	invalid resource directory name: /var/folders/_s/s0td900s19n0m1m1c2l0_p3nb7439n/T/resource_validator_tmp14646118543985275798/tmp-expanded/res navigation
```

### Feature requests: what underlying problem are you trying to solve with this feature?
Building our internal Android sample app that uses Jetpack Navigation.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.
Repro project: https://github.com/changusmc/android-navigation
This repro project is able to hit the same stack trace as my internal project.

Full error logs from building my project: [worker-1-AndroidResourceValidator.log](https://github.com/bazelbuild/bazel/files/3865870/worker-1-AndroidResourceValidator.log)

### What operating system are you running Bazel on?
MacOSX 10.15.1

### What's the output of `bazel info release`?
release 1.0.0-homebrew

###  Have you found anything relevant by searching the web?
Need to use aapt2 to build Jetpack Navigation: https://stackoverflow.com/a/57174533/3338374
Bazel should be defaulting `aapt2` by now and upcoming PRs will not allow `aapt1` https://github.com/bazelbuild/bazel/issues/10000

### Any other information, logs, or outputs that you want to share?


> Replace these lines with your answer.
>
> If the files are large, upload as attachment or provide link.

**Comment by changusmc** [comment]
> @jin asked me to comment here instead of filing a new issue. There's still one action that is using `aapt1` and when I build my project that uses Jetpack Navigation, it hits the code path for
> 
> https://github.com/bazelbuild/bazel/blob/c018f607c9b52d301841c4d0fd2d8771c711d299/src/tools/android/java/com/google/devtools/build/android/AndroidResourceValidatorAction.java#L148
> 
> I included a repro project below.
> 
> ### Description of the problem / feature request:
> The source of the problem stemmed building our app which has Jetpack Navigation. Running `bazel build //android/...` gives me the same error found in https://stackoverflow.com/a/57174533/3338374
> 
> ```
> 	invalid resource directory name: /var/folders/_s/s0td900s19n0m1m1c2l0_p3nb7439n/T/resource_validator_tmp14646118543985275798/tmp-expanded/res navigation
> ```
> 
> ### Feature requests: what underlying problem are you trying to solve with this feature?
> Building our internal Android sample app that uses Jetpack Navigation.
> 
> ### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.
> Repro project: https://github.com/changusmc/android-navigation
> This repro project is able to hit the same stack trace as my internal project.
> 
> Full error logs from building my project: [worker-1-AndroidResourceValidator.log](https://github.com/bazelbuild/bazel/files/3865870/worker-1-AndroidResourceValidator.log)
> 
> ### What operating system are you running Bazel on?
> MacOSX 10.15.1
> 
> ### What's the output of `bazel info release`?
> release 1.0.0-homebrew
> 
> ### Have you found anything relevant by searching the web?
> Need to use aapt2 to build Jetpack Navigation: https://stackoverflow.com/a/57174533/3338374
> Bazel should be defaulting `aapt2` by now and upcoming PRs will not allow `aapt1` #10000
> 
> ### Any other information, logs, or outputs that you want to share?
> > Replace these lines with your answer.
> > If the files are large, upload as attachment or provide link.

I was able to work around this by moving `res/navigation/nav_graph.xml` to `res/xml/nav_graph.xml`

**Comment by bchang** [comment]
This is a known issue, we have someone working on extricating aapt1 from AndroidResourceValidatorAction.

If I understand correctly, this isn't strictly blocking you from preparing your project for the incompatible change.

**Comment by changusmc** [comment]
> This is a known issue, we have someone working on extricating aapt1 from AndroidResourceValidatorAction.
> 
> If I understand correctly, this isn't strictly blocking you from preparing your project for the incompatible change.

I was able to side step this as mentioned above, but I don't know if it means that this project is ready for the incompatible change. I hope that once the validator action is moved to `aapt2`, I can move `nav_graph.xml` back to the expected `aapt2` path.

**Comment by bchang** [comment]
https://github.com/bazelbuild/bazel/commit/dbe76ceade3ea4561072fac9b4252b06293ee32a

The validator action was running both aapt1 and aapt2. With this commit, if you're using --incompatible_prohibit_aapt1, Bazel should not invoke aapt1.

**Comment by aehlig** [comment]
It seems the flag isn't flipped yet, so extending the migration window

**Comment by laurentlb** [comment]
Can you confirm this was released in 2.0?
If so, we can close the issue.
